src/multi_vector_simulator/C1_verification.py

```diff
@@ -83,6 +94,72 @@ def lookup_file(file_path, name):
         raise FileNotFoundError(msg)
 
 
+def find_value_by_key(data, target, result=None):
```

Comment:
maybe add this to `utils` instead?

Reply:
Done: 4ea6ada8093b00670566b1c4c8f18d48afd30f3d